lib/jsdom/living/nodes/HTMLLabelElement-impl.js

```diff
@@ -4,9 +4,62 @@ const HTMLElementImpl = require("./HTMLElement-impl").implementation;
 const closest = require("../helpers/traversal").closest;
 
 class HTMLLabelElementImpl extends HTMLElementImpl {
+  constructor(args, privateData) {
+    super(args, privateData);
+
+    this._eventDefaults.click = this._handleClick.bind(this);
+  }
+
   get form() {
     return closest(this, "form");
   }
+
+  _handleClick(event) {
+    if (this.hasAttribute('for')) {
+      const node = this.ownerDocument.getElementById(this.getAttribute('for'));
+      if (node && this._isLabelable(node)) {
+        this._dispatchClickEvent(node);
+      }
+    } else if (this.children.length) {
+      const kids = Array.prototype.slice.call(this.querySelectorAll('*'));
```

Comment:
Can you use domSymbolTree instead? E.g.

```js
} else {
  for (const descendant of domSymbolTree.treeIterator(this)) {
    if (this._isLabelable(descendant)) {
      this._dispatchClickEvent(descendant);
      break;
    }
  }
}
```

This will avoid query-selector, plus creating an array and then mutating it a bunch.